### PR TITLE
Improve the navigation documentation wording

### PR DESCRIPTION
**Pull request by ershn**
Changed the navigation documentation wording to make it a bit clearer (also fixed some typos).

Here is the detail of the changes
- make some sentences clearer by rewording or making them shorter
- remove unnecessary capital letters in page titles
- add some missing spaces
- add some commas to make sentence parsing easier
- fix some typos/english mistakes

Some changes are arguably pretty small but I think they make the documentation easier to read.
Don't hesitate to tell me if you think some changes are unnecessary though, I'll revert them back.

### CODE REVIEW

File: tutorials/navigation/navigation_debug_tools.rst [hunk 1]
```diff
@@ -1,56 +1,53 @@
...
-The navigation debug visualization is enabled by default inside the Editor.
-To visualize navigation meshes and connections also at runtime 
-enable the option ``Visible Navigation`` in the editor debug menu.
+The navigation debug visualizations are enabled by default inside the Editor.
+To visualize navigation meshes and connections at runtime too, enable the option ``Visible Navigation`` in the editor ``Debug`` menu.
 
 .. image:: img/navigation_debug_toggle.png
 
-In Godot debug builds the navigation debug can also be toggled on the NavigationServers from scripts.
+In Godot debug builds, the navigation debug can also be toggled through the NavigationServer singletons from scripts.
```
Comment (AThousandShips) [inline tutorials/navigation/navigation_debug_tools.rst:19]:
```suggestion
In Godot debug builds the navigation debug can also be toggled through the NavigationServer singletons from scripts.
```
This break isn't helping and breaks the flow IMO, this is different from "If X, then Y"

File: tutorials/navigation/navigation_introduction_2d.rst [hunk 2]
```diff
@@ -11,66 +11,68 @@ Godot provides the following objects and classes for 2D navigation:
...
 
-    The NavigationServer is best suited for 2D realtime gameplay that does require actors to reach any possible position within an navmesh defined area.
-    Meshbased navigation scales well with large gameworlds as a large area can often be defined with a single polygon when it would require many, many grid cells.
+    The NavigationServer is best suited for 2D realtime gameplay that does require actors to reach any possible position within a navmesh defined area.
+    Mesh-based navigation scales well with large game worlds as a large area can often be defined with a single polygon when it would require many, many grid cells.
 
     The NavigationServer holds different navigation maps that each consist of regions that hold navigation mesh data.
     Agents can be placed on a map for avoidance calculation.
-    RIDs are used to reference the internal maps, regions and agents when communicating with the server.
+    RIDs are used to reference internal maps, regions and agents when communicating with the server.
```
Comment (AThousandShips) [inline tutorials/navigation/navigation_introduction_2d.rst:24]:
```suggestion
    RIDs are used to reference internal maps, regions, and agents when communicating with the server.
```
Oxford comma

File: tutorials/navigation/navigation_introduction_2d.rst [hunk 3]
```diff
@@ -11,66 +11,68 @@ Godot provides the following objects and classes for 2D navigation:
...
+    - The NavigationServer2D will join the navigation meshes of regions by proximity for a combined navigation mesh.
 
 - :ref:`NavigationLink2D<class_NavigationLink2D>` Node
-    A Node that connects two positions on navigation mesh over arbitrary distances for pathfinding.
+    A Node that connects two positions on navigation meshes over arbitrary distances for pathfinding.
 
     - The link can be enabled / disabled.
     - The link can be made one-way or bidirectional.
-    - The use in pathfinding can be further restricted through the navigationlayers bitmask.
+    - The use in pathfinding can be further restricted through a navigationlayers bitmask.
```
Comment (AThousandShips) [inline tutorials/navigation/navigation_introduction_2d.rst:56]: This references the `navigation_layers` property so this is not a correct change

File: tutorials/navigation/navigation_introduction_3d.rst [hunk 4]
```diff
@@ -11,71 +11,69 @@ Godot provides the following objects and classes for 3D navigation:
...
+    - The NavigationServer3D will join the navigation meshes of regions by proximity for a combined navigation mesh.
 
 - :ref:`NavigationLink3D<class_NavigationLink3D>` Node
-    A Node that connects two positions on navigation mesh over arbitrary distances for pathfinding.
+    A Node that connects two positions on navigation meshes over arbitrary distances for pathfinding.
 
     - The link can be enabled / disabled.
     - The link can be made one-way or bidirectional.
-    - The use in pathfinding can be further restricted through the navigationlayers bitmask.
+    - The use in pathfinding can be further restricted through a navigationlayers bitmask.
```
Comment (AThousandShips) [inline tutorials/navigation/navigation_introduction_3d.rst:56]: Same here

File: tutorials/navigation/navigation_introduction_3d.rst [hunk 5]
```diff
@@ -11,71 +11,69 @@ Godot provides the following objects and classes for 3D navigation:
...
-    as a large area can often be defined with a single polygon when it would require many, many grid cells.
+    The NavigationServer is best suited for 3D realtime gameplay that does require actors to reach any possible position within a navmesh defined area.
+    Mesh-based navigation scales well with large game worlds as a large area can often be defined with a single polygon when it would require many, many grid cells.
 
-    The NavigationServer holds different navigation maps that each consist of regions that hold navigation mesh
-    data. Agents can be placed on a map for avoidance calculation. RIDs are used to reference the internal maps,
-    regions and agents when communicating with the server.
+    The NavigationServer holds different navigation maps that each consist of regions that hold navigation mesh data.
+    Agents can be placed on a map for avoidance calculation.
+    RIDs are used to reference internal maps, regions and agents when communicating with the server.
```
Comment (AThousandShips) [inline tutorials/navigation/navigation_introduction_3d.rst:24]:
```suggestion
    RIDs are used to reference internal maps, regions, and agents when communicating with the server.
```

File: tutorials/navigation/navigation_using_navigationmeshes.rst [hunk 6]
```diff
@@ -223,7 +225,7 @@ The following script creates a new 3D navigation region and fills it with proced
 Navmesh for 3D GridMaps
 ~~~~~~~~~~~~~~~~~~~~~~~
 
-The following script creates a new 3D navigation mesh for each GridMap items, clears the current grid cells and adds new procedual grid cells with the new navigation mesh.
+The following script creates a new 3D navigation mesh for each GridMap items, clears the current grid cells and adds new procedural grid cells with the new navigation mesh.
```
Comment (AThousandShips) [inline tutorials/navigation/navigation_using_navigationmeshes.rst:228]:
```suggestion
The following script creates a new 3D navigation mesh for each GridMap items, clears the current grid cells, and adds new procedural grid cells with the new navigation mesh.
```

File: tutorials/navigation/navigation_using_navigationmeshes.rst [hunk 7]
```diff
@@ -33,25 +37,23 @@ The outline polygons are later converted to real NavigationMesh resources for th
...
 
     The NavigationServer does not connect navigation mesh islands from the same NavigationMesh resource.
-    Do not create multiple disconnected islands in the same NavigationRegion2D and NavPoly resource if they should be later connected.
+    Do not create multiple disconnected islands in the same NavigationRegion2D or NavigationPolygon resource if they should be later connected.
 
-For 2D no similar navigation mesh baking with geometry parsing exists like in 3D.
-The Geometry2D class functions for offset, merge, intersect and clip can be used
-to shrink or enlarge existing NavigationPolygons to different actor sizes.
+For 2D, no similar navigation mesh baking with geometry parsing exists like in 3D.
+The Geometry2D class functions for offset, merge, intersect and clip can be used to shrink or enlarge existing NavigationPolygons for different actor sizes.
```
Comment (AThousandShips) [inline tutorials/navigation/navigation_using_navigationmeshes.rst:56]:
```suggestion
The Geometry2D class functions for offset, merge, intersect, and clip can be used to shrink or enlarge existing NavigationPolygons for different actor sizes.
```

File: tutorials/navigation/navigation_debug_tools.rst (hunk 1 above)
Comment (ershn) [reply tutorials/navigation/navigation_debug_tools.rst]: Fair enough. I removed the break.

File: tutorials/navigation/navigation_introduction_2d.rst (hunk 2 above)
Comment (ershn) [reply tutorials/navigation/navigation_introduction_2d.rst]: Fixed.

File: tutorials/navigation/navigation_introduction_2d.rst (hunk 3 above)
Comment (ershn) [reply tutorials/navigation/navigation_introduction_2d.rst]: Fixed.

File: tutorials/navigation/navigation_introduction_3d.rst (hunk 4 above)
Comment (ershn) [reply tutorials/navigation/navigation_introduction_3d.rst]: Fixed.

File: tutorials/navigation/navigation_introduction_3d.rst (hunk 5 above)
Comment (ershn) [reply tutorials/navigation/navigation_introduction_3d.rst]: Fixed.

File: tutorials/navigation/navigation_using_navigationmeshes.rst (hunk 6 above)
Comment (ershn) [reply tutorials/navigation/navigation_using_navigationmeshes.rst]: Fixed.

File: tutorials/navigation/navigation_using_navigationmeshes.rst (hunk 7 above)
Comment (ershn) [reply tutorials/navigation/navigation_using_navigationmeshes.rst]: Fixed.

**Comment by smix8** [comment]
Can not say much about the english part. The technical navigation part looks fine.

I would like to ask you to also change all uses of navmesh / navmeshes / navigationmesh / navigationmeshes to navigation mesh / navigation meshes in the other navigation pages. I actually had that planned for my own draft for a documentation update but I am glad someone beat me to it because I dont find the time to finish that.

The issue with those is not even the english but that plenty of users think 'navmesh' and 'navigationmesh' is something different or at least are confused about it. Also many users that confused 'navigationmesh' (as a general concept) with the 'NavigationMesh' resource in Godot. Same with 2D where many confuse the 2D 'NavigationPolygon' resource with 'navigation (mesh) polygon'. So I think having a unified term used across the documentation could help with that confusion.

Some of the older descriptions need to be brought up to speed but it is fine to just fix their wording for now.

**Comment by ershn** [comment]
Sure. I will do that.
I admit that was a bit confused by that at first too.

Another thing I noticed is that the documentation uses PascalCase for some concepts that the API doesn't expose as classes. For example in `Using NavigationMaps` NavigationMap is PascalCase but there is no NavigationMap class exposed anywhere (maybe it's defined in the implementation but even then I'm not sure the user manual should concern itself with that).
I think that changing the case would help in clearing some of the confusion in these situations.
Does that change seems ok to you ?

**Comment by smix8** [comment]
`NavigationMap` and `NavigationPath` want everyone to know that they will be late but still plan to join in one day. I couldn't get the NavigationPath in yet because when I opened the door to the NavigationMap all those little technical-debt monsters tried to crawl out.

**Comment by ershn** [comment]
Oh I see. I'll leave that part as is then.

**Comment by ershn** [comment]
I just pushed the changes. Please check if everything is ok.

- unify navmesh(es) / navigationmesh(es) to navigation mesh(es)
- change `NavigationMesh` to `navigation mesh` in the following scenarios
  - `NavigationMesh` referred to 2D meshes, or both 2D and 3D meshes
  - `navigation mesh` is used in the exact same spot in the corresponding 2D documentation (like section titles)
  - (I double checked everything but maybe I got some of them wrong. Please check.)
- unify navpoly to navigation polygon
- change `navigationlayer` to either `navigation layer` (the concept) or `navigation_layers` (the property) depending on context
- fix small english errors, add some commas where it makes things easier to parse

**Review by smix8 — APPROVED** [review]
Looks fine, thanks.

**Review by mhilbrunner — APPROVED** [review]

**Comment by mhilbrunner** [comment]
Great work, thanks! Merged.

**Pull request merged**